### DISCUSSION TITLE
What are the keycodes for F13-F24 keys?

### DISCUSSION THREAD

**Discussion by F1reP0wer12345** [AntiMicroX/antimicrox#837] [Q&A]
Simple question. I don't have a keyboard with these keys, but i would like to use my controller to simulate these keys so they can later be used for example for launching a program. Because almost no program or keyboard uses these keys, they can be used to perform actions and not get in the way of my regular keyboard shortcuts. Thanks in advance!

**Comment by LAPIII** [comment]
Do you mean Virtual key codes? Find them [here](https://learn.microsoft.com/en-us/windows/win32/inputdev/virtual-key-codes).

**Reply by F1reP0wer12345** [reply]
No, sorry, that's not what I meant. AntiMicroX has a key checker. In this key checker i can press any button on my keyboard and get 3 different values in return. For example, my right CTRL key has the values `0xa3`, `0x1000021` and `0x11000021`. The first one called `native key value`, the second is called `QT Key Value`, and the final is the `AntiMicroX Key Value`. I'm looking for the AntiMicroX key values for F13-F24. As I said before, almost no program uses these keys, and i want to press the button, then intercept it, then run a bash script. The best solution would be to run bash code directly, but since AntiMicroX can only press a button when another is pressed, this was the best I could come up with. However without a keyboard with the keys F13-F24, i can't run the key checker and get the values of these keys. So i was hoping someone could help me by pressing these buttons inside the key checker, and then give me the values as a reply. Though I'm sure that looking through the source code could help me as well. I was able to find some reference to F24 - F1 if I'm not mistaken, but because of my little C++ knowledge, I don't know how to find the value I'm looking for. Here's what i found: https://github.com/AntiMicroX/antimicrox/blob/557dcad2cd99dc6468534cb783a54df33e9826b9/src/qtwinkeymapper.cpp#L231
Thanks for sharing the Virtual key codes though. I hope you or anyone for that matter can help me with this. Thanks in advance.

**Reply by LAPIII** [reply]
Those keys are `Shift + F key` .
